adoc/chapters/programming_interface.adoc

```diff
@@ -10183,6 +10183,12 @@ eventually be deallocated with [code]#sycl::free# in order to avoid a memory
 leak.  If there are not enough resources to allocate the requested memory,
 these functions return [code]#nullptr#.
 
+When the allocation size is zero bytes ([code]#numBytes# or [code]#count# is
+zero), these functions behave in a manor consistent with {cpp}
+[code]#std::malloc#.  The value returned is unspecified in this case, and the
+returned pointer may not be used to access storage.  If this pointer is not
+null, it must be passed to [code]#sycl::free# to avoid a memory leak.
```

Comment:
```suggestion
When the allocation size is zero bytes ([code]#numBytes# or [code]#count# is
zero), these functions returns a null pointer.

Note that this behavior is still compatible with the {cpp}
[code]#std::malloc# as [code]#sycl::free# can be used without any effect
on the returned (null) pointer.
```

Reply:
As mentioned a few hours ago at the meeting you and I were at, that behavior is NOT specification compatible with the C or C++ versions of malloc.

Code that assumes that `malloc(0)` always returns a null pointer value (and, say, doesn't call `free` on it) cannot be backported to regular C or C++.

I REJECT that characterization in the wording, and I REJECT any changes to the wording until the vote takes place next week.

Reply:
Ah, this is an important new point of information to me.
You do not only want that some typical C/C++ `std::malloc`/`std::free` can be used in a USM SYCL context but you also want that a code developed specifically in a USM SYCL context can be used, almost as is, back into a C/C++ context.
Interesting.
I am unsure this is a realistic use case, but at least this is to be considered.